docs/decisions/0005-credly-xblock.rst

```diff
@@ -0,0 +1,286 @@
+-  received_award: Boolean to store if the user received a badge for
+   this sub-section.
+-  check_earned: True if they are eligible for a badge.
+-  assertion_url: The URL of the asserted badge in case it was already
+   awarded.
+-  award_message: Message the user will see upon receiving a badge.
+-  motivation_message: Message the user will see if they do not qualify
+   for a badge.
+
+Once setup, this XBlock will display a button with the text “Click here
+to view your results.”, once the learner clicks on it, the JavaScript
+function getGrades will be executed to request the current grading of
+the setup sub-section, if the minimum pass mark is achieved, it will
+execute the method new_award_badge using the XBlock handlerUrl runtime
+to award a new badge using the current setup badging service set up on
+the edx-platform, and reload the XBlock HTML to display the newly
+created badge for this user, otherwise, it will reload the HTML to an
+error message.
+
+XBlock Proposal
+===============
+
+Given how badgr-xblock can interact with Badgr and the edx-platform to
+award badges, we could create a new XBlock to replicate this exact
+functionality using Credly as a backend service for badging to award
+badges on sub-section completion. A template for this XBlock could be
+generated using the XBlock SDK, this SDK also contains various examples
+of how we could use XBlock for various scenarios.
+
+We could query course grades using the grades v1 API grade book
+endpoint, which will return a list of the scores for each graded
+subsection per user, another option could be using the XBlock get_parent
+to trace the parent block to get a problem using an usage_key setup on
+this XBlock, to award badges for a specific unit or problem, the
+disadvantage of this method would be that the XBlock can only get the
+grade of the unit or problem at the moment this XBlock is being
+rendered. A mix of these methods could also be used to award a badge
+based on the completion of a group of sub-sections or a group of
+usage_keys which are graded problems.
+
+As described before, there are various methods we could use to retrieve
+information from the XBlock or platform to determine the conditions to
+generate a badge assertion (award a badge), either by requesting data
+using any API endpoint on the edx-platform using JavaScript, accessing
+the environment using the XBlock runtime API, using information stored
+in any of the scopes of the XBlock, using information stored in the
+XBlock fields, or using any of the XBlock services, such as the user
+service. More information on what API methods are available can be found
+in the XBlock API guide.
+
+Bading with XBlock Cons
+-----------------------
+
+- Course/subsection scope which means there would be no badges that
+  involve multiple courses, for example, courses of the program.
+- Badges can only be awarded manually when a learner interacts
+  with an XBlock.
```

Comment:
One more:

This xblock will rely on the Credly backend implementation, so the integration with Credly will depend upon the Credly backend implementation and where where it will be located (Plugin, add a new Djangoapp in edx-platform, Python package...). @kuipumu

Reply:
Since we will be merging https://agile-jira.pearson.com/browse/PADV-233 and https://agile-jira.pearson.com/browse/PADV-234 into a single document, this is explained in PADV-234.